### PR TITLE
CA1068: allow excluded_symbol_names user option

### LINKED ISSUE #4467: Cancellation tokens passed to constructors
**Diagnostic ID**: [CA1068](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1068)

Would it make sense for this rule to ignore constructors on the premise that constructors are more likely to be assigning the cancellation token to a field than using it?

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #4467

### CODE REVIEW

File: docs/Analyzer Configuration.md [hunk 1]
```diff
@@ -264,6 +264,7 @@ Option Name: `excluded_symbol_names`
 Configurable Rules:
 [CA1001](https://docs.microsoft.com/visualstudio/code-quality/ca1001),
 [CA1062](https://docs.microsoft.com/visualstudio/code-quality/ca1062),
+[CA1068](https://docs.microsoft.com/visualstudio/code-quality/ca1068),
```
Comment (Youssef1313) [inline docs/Analyzer Configuration.md:267]: The docs needs to be updated to mention the configurability of the rule done in this PR, in addition to the recent configurability change done with https://github.com/dotnet/roslyn-analyzers/issues/2851 for the same rule.

File: src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLastTests.cs [hunk 2]
```diff
@@ -513,6 +513,72 @@ Imports System.Threading
...
+{
+    public " + prefix + "C1" + suffix + @"(CancellationToken t, int i) {}
+
+    public " + prefix + "C1" + suffix + @"(CancellationToken t, float f) {}
+}
+
+public class C2
+{
+    public " + prefix + "C2" + suffix + @"(CancellationToken t, int i) {}
+}"
```
Comment (Youssef1313) [inline src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLastTests.cs:556]: Could you add a test for:

```csharp
public record R(CancellationToken t, int i) {}
```

File: docs/Analyzer Configuration.md (hunk 1 above)
Comment (mavasani) [reply docs/Analyzer Configuration.md:267]: Thanks @Youssef1313 - agreed.

**Review by mavasani — APPROVED** [review]
LGTM.

Kindly update the docs with a doc PR OR file an issue on dotnet/docs repo to update the documentation.

**Comment by mavasani** [comment]
@Evangelink 6.0.1xx-preview1 branch is primarily created for new analyzers that we do not want to include in 5.x NuGet package releases for NetAnalyzers package. Any bug fixes or enhancements to existing analyzers can target master.

File: eng/Versions.props [hunk 3]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <VersionPrefix>3.3.2</VersionPrefix>
     <PreReleaseVersionLabel>beta1</PreReleaseVersionLabel>
-    <NetAnalyzersVersionPrefix>5.0.1</NetAnalyzersVersionPrefix>
+    <NetAnalyzersVersionPrefix>6.0.0</NetAnalyzersVersionPrefix>
```
Comment (Evangelink) [inline eng/Versions.props:5]:
```suggestion
    <NetAnalyzersVersionPrefix>5.0.1</NetAnalyzersVersionPrefix>
```

File: docs/Analyzer Configuration.md (hunk 1 above)
Comment (Evangelink) [reply docs/Analyzer Configuration.md:267]: https://github.com/dotnet/docs/pull/21853

**Review by mavasani — APPROVED** [review]

**Comment by mavasani** [comment]
@Evangelink Can you please resolve merge conflicts?

File: docs/Analyzer Configuration.md [hunk 4]
```diff
@@ -262,48 +262,49 @@ Example: `dotnet_code_quality.try_determine_additional_string_formatting_methods
...
-[CA5380](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca5380),
-[CA5381](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca5381),
-[CA5382](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca5382),
-[CA5383](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca5383),
-[CA5384](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca5384),
-[CA5387](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca5387),
-[CA5388](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca5388),
-[CA5389](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca5389),
-[CA5390](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca5390)
+[CA1001](https://docs.microsoft.com/visualstudio/code-quality/ca1001),
```
Comment (Youssef1313) [inline docs/Analyzer Configuration.md:265]: Changes here reverts #4515

Comment (Evangelink) [reply docs/Analyzer Configuration.md:265]: Sorry moved to fast on the conflict... I will fix that now. Thanks for the check!

**Comment by mavasani** [comment]
@Evangelink There are more merge conflicts :-)

**Comment by Evangelink** [comment]
Thank you @mavasani!

**Pull request merged**